### PR TITLE
Table panel: Adds text-align to stock Table panel

### PR DESCRIPTION
**Pull request by teodoryantcheff**
This adds text-align option to the stock Table panel.

Ref:
https://community.grafana.com/t/right-align-numbers-in-table/13261/3
https://community.grafana.com/t/column-aligning-in-tables/13748

Fixes #
#9544

**Special notes for your reviewer:** First time submitting a PR... have mercy.

### CODE REVIEW

**Review by NellaFej — APPROVED** [review]
Approved

**Comment by teodoryantcheff** [comment]
fake activity

**Comment by teodoryantcheff** [comment]
Hey, anything on my end I can do to help merging this PR in?

**Comment by hugohaggmark** [comment]
> Hey, anything on my end I can do to help merging this PR in?

Hi @teodoryantcheff and thanks for contributing, firstly you should change the `merge` target to `master`. It's currently set to `v.6.4.x`.

File: packages/grafana-ui/src/components/Table/TableCellBuilder.tsx [hunk 1]
```diff
@@ -58,6 +58,8 @@ export interface ColumnStyle {
   valueMaps?: any;
   rangeMaps?: any;
 
+  align?: string;
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Table/TableCellBuilder.tsx]: Consider typing `align` as  'left' | 'center' | 'right' | 'justify'
```suggestion
  align?: 'default' | 'left' | 'center' | 'right' | 'justify';
```

File: public/app/plugins/panel/table/column_options.ts [hunk 2]
```diff
@@ -44,6 +45,14 @@ export class ColumnOptionsCtrl {
       { text: 'YYYY-MM-DD', value: 'YYYY-MM-DD' },
     ];
     this.mappingTypes = [{ text: 'Value to text', value: 1 }, { text: 'Range to text', value: 2 }];
+    this.alignTypes = [
+      { text: 'default', value: '' },
+      { text: 'left', value: 'left' },
+      { text: 'center', value: 'center' },
+      { text: 'right', value: 'right' },
+      { text: 'justify', value: 'justify' },
+      { text: 'initial', value: 'initial' },
```
Comment (hugohaggmark) [inline public/app/plugins/panel/table/column_options.ts]: Not sure if `initial` is something we want in this context? Shouldn't default cover this?

File: public/app/plugins/panel/table/column_options.ts [hunk 3]
```diff
@@ -78,6 +87,7 @@ export class ColumnOptionsCtrl {
       dateFormat: 'YYYY-MM-DD HH:mm:ss',
       thresholds: [],
       mappingType: 1,
+      align: '',
```
Comment (hugohaggmark) [inline public/app/plugins/panel/table/column_options.ts]: Maybe this should be more explicit like 'default' instead?

File: public/app/plugins/panel/table/renderer.ts [hunk 4]
```diff
@@ -264,6 +264,15 @@ export class TableRenderer {
       cellClasses.push('table-panel-cell-pre');
     }
 
+    if (column.style && column.style.align) {
+      cellStyles.push(`text-align:${column.style.align}`);
```
Comment (hugohaggmark) [inline public/app/plugins/panel/table/renderer.ts]: If the user has chosen to format as `Number` and not chosen any `alignment` then maybe the `default` should be `right`? wdyt @torkelo?

File: public/app/plugins/panel/table/renderer.ts [hunk 5]
```diff
@@ -228,17 +227,18 @@ export class TableRenderer {
...
 
     if (this.colorState.cell) {
-      cellStyle = ' style="background-color:' + this.colorState.cell + '"';
+      cellStyles.push('background-color:' + this.colorState.cell);
       cellClasses.push('table-panel-color-cell');
       this.colorState.cell = null;
     } else if (this.colorState.value) {
-      textStyle = ' style="color:' + this.colorState.value + '"';
+      cellStyles.push('color:' + this.colorState.value);
+      // console.log('styles:', cellStyles);
```
Comment (hugohaggmark) [inline public/app/plugins/panel/table/renderer.ts]: Please remove this comment

File: public/app/plugins/panel/table/renderer.ts [hunk 6]
```diff
@@ -264,6 +264,15 @@ export class TableRenderer {
       cellClasses.push('table-panel-cell-pre');
     }
 
+    if (column.style && column.style.align) {
+      cellStyles.push(`text-align:${column.style.align}`);
+    }
+
+    if (cellStyles.length) {
+      cellStyle = ' style="' + cellStyles.join(';') + '"';
+      // console.log('style:', cellStyle);
```
Comment (hugohaggmark) [inline public/app/plugins/panel/table/renderer.ts]: Please remove this comment

**Review by hugohaggmark — CHANGES_REQUESTED** [review]
Great work on this @teodoryantcheff and thanks again for taking the time to contribute, I've left some minor questions for you to take a look at. 

Also I'm missing tests for this change. Take a look in `renderer.test.ts` for inspiration on how to test this.

One last thing, for existing dashboards/table panels the align select/drop down doesn't seem to work but for new table panels it works:
table-align

I think you'll have to add a row in `DashboardMigrator` for migrating old `table panels`.

File: packages/grafana-ui/src/components/Table/TableCellBuilder.tsx (hunk 1 above)
Comment (teodoryantcheff) [reply packages/grafana-ui/src/components/Table/TableCellBuilder.tsx]: Aggreed.

File: public/app/plugins/panel/table/column_options.ts (hunk 3 above)
Comment (teodoryantcheff) [reply public/app/plugins/panel/table/column_options.ts]: 'default' it is.

File: public/app/plugins/panel/table/renderer.ts (hunk 4 above)
Comment (teodoryantcheff) [reply public/app/plugins/panel/table/renderer.ts]: Not changing this for the time being.

File: public/app/plugins/panel/table/renderer.ts (hunk 6 above)
Comment (teodoryantcheff) [reply public/app/plugins/panel/table/renderer.ts]: done.

File: public/app/plugins/panel/table/column_options.ts (hunk 2 above)
Comment (teodoryantcheff) [reply public/app/plugins/panel/table/column_options.ts]: Just added all the css styles there are.

> One last thing, for existing dashboards/table panels the align select/drop down doesn't seem to work but for new table panels it works:
> table-align

Had (and still have) no idea how to make it proper.

**Comment by teodoryantcheff** [comment]
Think all comments are resolved in https://github.com/grafana/grafana/pull/21175

**Pull request closed without merging**